### DISCUSSION TITLE
Customize DomainException class

### DISCUSSION THREAD

**Discussion by shashikantpawar21** [Eventuous/eventuous#145] [Q&A]
Currently DomainException only accepts error message field, 

How can I make a my own customize DomainException class and add an extra field called error type that is required by application to handle UI behaviour as per the error type.. 

A quick help would be really appreciated.

**Comment by alexeyzimarev** [comment]
Use inheritance. I don't think the `DomainException` class is sealed.

**Reply by alexeyzimarev** [reply]
If you need some stuff to be passed over to the UI using `ErrorResult`, please elaborate it more explicitly, so I can better understand what you want to do.

**Reply by shashikantpawar21** [reply]
no the above thing worked

**Reply by alexeyzimarev** [reply]
Sorry man, but you aren't helping, really. I asked you to elaborate on what you want to do, and you replied "nothing works". That's quite ignorant.

**Reply by bartelink** [reply]
>no the above thing worked

@shashikantpawar21 Did you mean that the first suggestion worked?

If so, I'd suggest:
- including the word "thanks" as a signal of a) the problem being resolved b) your appreciation for the time taken
- marking the answer and/or closing the discussion (not sure what you can do because I have not created a discussion myself)

Investing the time in communicating clearly is always worth it, but especially when the maintainer of a project is taking the time to answer your questions.

While I appreciate it may not be your first language, small things like capital letters and punctuation also signal you're participating in a discussion, as opposed to coming across as being entitled (I'm not assuming you are, but terse communications can often be misinterpreted as that)
